pkg/netutils/ips.go

```diff
@@ -8,15 +8,18 @@ import (
 	"github.com/replicatedhq/embedded-cluster/pkg/runtimeconfig"
 )
 
+// Dependency injection variables for testing
+var (
+	networkInterfaceProvider = DefaultNetworkInterfaceProvider
+	tryDiscoverPublicIP      = runtimeconfig.TryDiscoverPublicIP
+)
+
 // adapted from https://github.com/k0sproject/k0s/blob/v1.30.4%2Bk0s.0/internal/pkg/iface/iface.go#L61
 func FirstValidAddress(networkInterface string) (string, error) {
 	ipnet, err := FirstValidIPNet(networkInterface)
 	if err != nil {
 		return "", fmt.Errorf("get ipnet for interface %s: %w", networkInterface, err)
 	}
-	if ipnet.IP.To4() == nil {
-		return "", fmt.Errorf("interface %s has no ipv4 addresses", networkInterface)
-	}
```

Comment:
This isn't required and will be unreachable since `FirstValidIPNet` will take care of filtering non ipv4 interfaces, see:
- https://github.com/replicatedhq/embedded-cluster/blob/6ed500f2adc19ef8a3da6ce4f43c78afb7ccb73e/pkg/netutils/ips.go#L111-L115